tests/functions/module_filter.test.ts

```diff
@@ -217,3 +217,787 @@ describe("UNIQUE function", () => {
+      [1, 1, 1],
+      [1, 3, 3],
+      [2, 1, 1],
+      [2, 1, 1],
+      [2, 1, 2],
+      ["", "", ""],
+    ]);
+  });
+
+  test("Full sorting with multiple columns using default parameters", () => {
+    //prettier-ignore
+    const grid = {
+       A1: "1",  B1: "1",  C1: "1",
+       A2: "2",  B2: "1",  C2: "2",
+       A3: "3",  B3: "2",  C3: "1",
+       A4: "1",  B4: "3",  C4: "3",
+       A5: "2",  B5: "3",  C5: "1",
+       A6: "4",  B6: "2",  C6: "1",
+       A7: "3",  B7: "4",  C7: "4",
+       A8: "2",  B8: "1",  C8: "1",
+       A9: "4",  B9: "1",  C9: "2",
+      A10: "2", B10: "1", C10: "1",
+    };
+    const model = createModelFromGrid(grid);
+    setCellContent(model, "A11", "=SORTN(A1:C10)");
+    expect(getRangeValuesAsMatrix(model, "A11:C12")).toEqual([
+      [1, 1, 1],
+      ["", "", ""],
+    ]);
+  });
+
+  test("Full sorting with different display ties modes", () => {
+    //prettier-ignore
+    const grid = {
+       A1: "1",  B1: "1",  C1: "1",
+       A2: "2",  B2: "1",  C2: "2",
+       A3: "3",  B3: "2",  C3: "1",
+       A4: "1",  B4: "3",  C4: "3",
+       A5: "2",  B5: "3",  C5: "1",
+       A6: "2",  B6: "3",  C6: "1",
+       A7: "3",  B7: "4",  C7: "4",
+       A8: "2",  B8: "1",  C8: "1",
+       A9: "4",  B9: "1",  C9: "2",
+      A10: "2", B10: "1", C10: "1",
+    };
+    const model = createModelFromGrid(grid);
+    setCellContent(model, "A11", "=SORTN(A1:C10, 3, 1)");
+    expect(getRangeValuesAsMatrix(model, "A11:C15")).toEqual([
+      [1, 1, 1],
+      [1, 3, 3],
+      [2, 1, 1],
+      [2, 1, 1],
+      ["", "", ""],
+    ]);
+    setCellContent(model, "A11", "=SORTN(A1:C10, 5, 2)");
+    expect(getRangeValuesAsMatrix(model, "A11:C16")).toEqual([
+      [1, 1, 1],
+      [1, 3, 3],
+      [2, 1, 1],
+      [2, 1, 2],
+      [2, 3, 1],
+      ["", "", ""],
+    ]);
+    setCellContent(model, "A11", "=SORTN(A1:C10, 5, 3)");
+    expect(getRangeValuesAsMatrix(model, "A11:C18")).toEqual([
+      [1, 1, 1],
+      [1, 3, 3],
+      [2, 1, 1],
+      [2, 1, 1],
+      [2, 1, 2],
+      [2, 3, 1],
+      [2, 3, 1],
+      ["", "", ""],
+    ]);
+  });
+
+  test("Sorting with bad 'n' argument", () => {
+    //prettier-ignore
+    const grid = {
+       A1: "1",  B1: "1",  C1: "1",
+       A2: "2",  B2: "1",  C2: "2",
+       A3: "3",  B3: "2",  C3: "1",
+       A4: "1",  B4: "3",  C4: "3",
+       A5: "2",  B5: "3",  C5: "1",
+       A6: "4",  B6: "2",  C6: "1",
+       A7: "3",  B7: "4",  C7: "4",
+       A8: "2",  B8: "1",  C8: "1",
+       A9: "4",  B9: "1",  C9: "2",
+      A10: "2", B10: "1", C10: "1",
+    };
+    const model = createModelFromGrid(grid);
+    setCellContent(model, "A11", "=SORTN(A1:C10, -1, 0)");
+    expect(getRangeValuesAsMatrix(model, "A11")).toEqual([["#ERROR"]]);
+    expect(checkFunctionDoesntSpreadBeyondRange(model, "A11")).toBeTruthy();
+  });
+
+  test("Sorting with a too big 'n' value returns full sorted range", () => {
+    //prettier-ignore
+    const grid = {
```

Comment:
all those tests are ok with 10 rows. But less would have also worked. As a general advice, having the minimal setup to describe a behavior is best